### ISSUE TITLE
Constant crash on login on 2.28.1 with `Segmentation fault: 11`

### ISSUE THREAD

**Issue by churik** [status-im/status-desktop#14643]
# Bug Report

## Description
Cannot use certain accounts on 2.28.1

## Steps to reproduce
- Attempted login with a previously used user account.
- Experienced an initial crash on the first attempt (same error), successfully logged in after the initial crash.
- Disabled the "Archive Protocol enabled" option (suspected of causing an error)
- Encountered constant crashes upon attempting to log in again, preventing access.

#### Expected behavior
can login 

#### Actual behavior
```
INF 2024-05-08 16:23:54.432+02:00 mailserver not working                     topics="mailservers-service" tid=14581093 file=service.nim:161
INF 2024-05-08 16:23:54.432+02:00 removing active mailserver                 topics="mailservers-service" tid=14581093 file=service.nim:149
2024-05-08T16:23:54+0200 unset [github.com/anacrolix/torrent.(*Client).NewAnacrolixDhtServer.func2:406]: dht server on [::]:62728 (node id cef8087d8d3eedcb4e7d7798ab749fca043c0175) completed bootstrap ({NumAddrsTried:8 NumResponses:0})
2024-05-08T16:23:54.829+0200	INFO	communities/manager.go:382	failed to request missing messages	{"error": "stream reset"}
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Segmentation fault: 11
```
[geth.log](https://github.com/status-im/status-desktop/files/15250239/geth.log)

### Additional Information

- Status desktop version: [2.28.1](https://ci.infra.status.im/job/status-desktop/job/release/job/release%2F2.28.x/70/)
- Operating System: Mac OS Sonoma

**Comment by friofry** [comment]
I have a hunch: 
The sigsegv could be related to access to the unitised TorrentClient. `DisableCommunityHistoryArchiveProtocol` calls StopTorrentClient (`m.torrentClient = null`)

I have an open PR for a similar problem:
https://github.com/status-im/status-go/pull/5125#discussion_r1593650620

**Comment by friofry** [comment]
I was able to reproduce the above problem simply by :
- creating a new account
- applying the settings from the previous message
- restarting the application 
- cannot log in (torrent config has Enabled = true, and empty data dirs)

@MishkaRogachev We are probably working on the same components (#14235). I also have a problem with malformed torrent config

**Comment by MishkaRogachev** [comment]
> @MishkaRogachev We are probably working on the same components (#14235). I also have a problem with malformed torrent config

An issue I'm working on is a little different, but I'm experimenting a lot with the history archive. I'll post an update if I manage to reproduce a similar problem.

**Comment by friofry** [comment]
Narrowed down the reproduce steps:
1. create new account
2. enable archive protocol 
3. switch off `Debug` flag

On step 3. The torrent config gets corrupted. Because local version of nodeConfiguration.TorrentConfig is empty on Nim side (while on status go side it has valid data). When we SaveSettings it corrupts the node_config in database.

I've added node_config.Validate() before saving to database and now it fails

added the following test:

```
func TestConfigValidate(t *testing.T) {
	// GIVEN
	db, stop := setupTestDB(t)
	defer stop()

	nodeConfig := randomNodeConfig()
	err := nodeConfig.Validate()
	require.NoError(t, err)
	require.NoError(t, nodecfg.SaveNodeConfig(db, nodeConfig))

	// WHEN
	dbNodeConfig, err := nodecfg.GetNodeConfigFromDB(db)
	require.NoError(t, err)

	// THEN
	err = dbNodeConfig.Validate()
	require.NoError(t, err)
}
```
it fails with:

```
Error:      	Received unexpected error:
	Key: 'NodeConfig.KeycardPairingDataFile' Error:Field validation for 'KeycardPairingDataFile' failed on the 'required' tag
	Key: 'NodeConfig.LogLevel' Error:Field validation for 'LogLevel' failed on the 'eq=ERROR|eq=WARN|eq=INFO|eq=DEBUG|eq=TRACE' tag

```

And indeed this fields are required/validated in node_config.go:

```
KeycardPairingDataFile string validate:"required" to config.go 
```

**Comment by friofry** [comment]
Another issue with SaveSettings.

The TorrentConfig struct in dto/node_config.nim is:
```
  TorrentConfig* = object
    Enabled*: bool
```
and in `config.go` it's:
```
// TorrentConfig provides configuration for the BitTorrent client used for message history archives.
type TorrentConfig struct {
	// Enabled set to true enables Community History Archive protocol
	Enabled bool
	// Port number which the BitTorrent client will listen to for conntections
	Port int
	// DataDir is the file system folder Status should use for message archive torrent data.
	DataDir string
	// TorrentDir is the file system folder Status should use for storing torrent metadata files.
	TorrentDir string
}
```
It's converted to json and unmarshalled in value_handlers.go `NodeConfigHandler` and some fields are missing.
I can either mark them as required and add them on the nim side. (Or change the logic and make SaveSettings override only the values that are explicitly set. Shady logic)

**Comment by friofry** [comment]
Quick update. I found a better approach in `backend/node_config.nim`, which calls status-go methods to update only the required fields. `EnableCommunityHistoryArchiveSupport`

And `saveNodeConfiguration` from `settings/service.nim` should be avoided. It will always break NodeConfig.

**Comment by churik** [comment]
FYI: on latest master (17/06) on the same account this issue is still reproducible to me.
```
ERR 2024-06-17 16:00:29.871+02:00 error:                                     topics="accounts-service" tid=41948855 file=service.nim:655 procName=verifyDatabasePassword errDesription="failed to set `journal_mode` pragma: file is not a database"
ERROR[06-17|16:00:30.051|github.com/status-im/status-go/api/geth_backend.go:434] failed to initialize wallet db           package=status-go/api.GethStatusBackend err="failed to set `journal_mode` pragma: file is not a database"
ERROR[06-17|16:00:30.051|github.com/status-im/status-go/mobile/status.go:319]    loginAccount failed                      error="failed to set `journal_mode` pragma: file is not a database"
ERR 2024-06-17 16:00:30.052+02:00 node.login                                 topics="signals-manager" tid=41948855 file=signals_manager.nim:54 error="failed to set `journal_mode` pragma: file is not a database"
ERR 2024-06-17 16:00:30.053+02:00 login error                                topics="app-controller" tid=41948855 file=module.nim:419 methodName=onNodeLogin errDesription="failed to set `journal_mode` pragma: file is not a database"
Warning: QQuickItem::stackAfter: Cannot stack AccountMenuItemPanel_QMLTYPE_3358(0x6000036c8dc0, parent=0x6000036f7de0, geometry=0,0 342x64) after AccountMenuItemPanel_QMLTYPE_3358(0x6000036e53f0), which must be a sibling (:0, )
ERROR[06-17|16:00:39.173|github.com/status-im/status-go/api/geth_backend.go:434] failed to initialize wallet db           package=status-go/api.GethStatusBackend err="failed to set `journal_mode` pragma: file is not a database"
failed to set `journal_mode` pragma: file is not a database
ERR 2024-06-17 16:00:39.173+02:00 error:                                     topics="accounts-service" tid=41948855 file=service.nim:655 procName=verifyDatabasePassword errDesription="failed to set `journal_mode` pragma: file is not a database"
ERROR[06-17|16:00:39.357|github.com/status-im/status-go/api/geth_backend.go:434] failed to initialize wallet db           package=status-go/api.GethStatusBackend err="failed to set `journal_mode` pragma: file is not a database"
ERROR[06-17|16:00:39.357|github.com/status-im/status-go/mobile/status.go:319]    loginAccount failed                      error="failed to set `journal_mode` pragma: file is not a database"
ERR 2024-06-17 16:00:39.358+02:00 node.login                                 topics="signals-manager" tid=41948855 file=signals_manager.nim:54 error="failed to set `journal_mode` pragma: file is not a database"
ERR 2024-06-17 16:00:39.359+02:00 login error                                topics="app-controller" tid=41948855 file=module.nim:419 methodName=onNodeLogin errDesription="failed to set `journal_mode` pragma: file is not a database"
ERROR[06-17|16:00:47.448|github.com/status-im/status-go/api/geth_backend.go:434] failed to initialize wallet db           package=status-go/api.GethStatusBackend err="failed to set `journal_mode` pragma: file is not a database"
failed to set `journal_mode` pragma: file is not a database
ERR 2024-06-17 16:00:47.448+02:00 error:                                     topics="accounts-service" tid=41948855 file=service.nim:655 procName=verifyDatabasePassword errDesription="failed to set `journal_mode` pragma: file is not a database"
WARN [06-17|16:00:47.821|github.com/status-im/status-go/api/geth_backend.go:724] fleet is not supported, overriding with default value package=status-go/api.GethStatusBackend fleet=status.prod defaultFleet=shards.test
WRN 2024-06-17 16:00:47.865+02:00 Error decoding signal                      topics="signals-manager" tid=41948855 file=signals_manager.nim:49 err="Unknown signal received: mediaserver.started"
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Segmentation fault: 11
```

I'll reopen the issue.

**Comment by jrainville** [comment]
@friofry was the fix you did only prevent that type of crash from happening again or it would also unstuck people like @churik ?

If it was the former, how big of an effort is it to make it fix broken states? If it's too hard, maybe you can just help her unblock it manually, since I don't think other people got this situation.

**Comment by churik** [comment]
Attaching geth.log for this user: 
[geth.log](https://github.com/user-attachments/files/15902031/geth.log)

**Comment by friofry** [comment]
> was the fix you did only prevent that type of crash from happening again or it would also unstuck people like
It turned out I fixed a non-related crash.

But it's hard to tell what's wrong with the new crash without a database or steps to reproduce it.

The error in the original logs doesn't look the same as in the new config.
```
INFO [06-19|16:03:43.785|github.com/status-im/status-go/protocol/messenger.go:363]                       failed to find Waku service; trying WakuV2 error="waku service is not available"
```
and in the original log this line is also there, but 5 seconds before the crash.

@jrainville 
Yes, it would be hard to debug this without a hint. We've discussed with @churik that this test account is not critical.
I suggest to close it with "cannot reproduce" status for now until there will be a steps to reproduce.
